reflex/app.py

```diff
@@ -641,6 +646,67 @@ def compile(self):
         thread_pool.close()
         thread_pool.join()
 
+    @contextlib.asynccontextmanager
+    async def modify_state(self, token: str) -> AsyncIterator[State]:
+        """Modify the state out of band.
+
+        Args:
+            token: The token to modify the state for.
+
+        Yields:
+            The state to modify.
+
+        Raises:
+            RuntimeError: If the app has not been initialized yet.
+        """
+        if self.event_namespace is None:
+            raise RuntimeError("App has not been initialized yet.")
+        async with self.state_manager.modify_state(token) as state:
+            yield state
+            delta = state.get_delta()
+            if delta:
+                state._clean()
+                await self.event_namespace.emit_update(
+                    update=StateUpdate(delta=delta),
+                    sid=state.get_sid(),
+                )
+
+    def _process_background(self, state: State, event: Event) -> asyncio.Task | None:
+        """Process an event in the background and emit updates as they arrive.
+
+        Args:
+            state: The state to process the event for.
+            event: The event to process.
+
+        Returns:
+            Task if the event was backgroundable, otherwise None
+        """
+        substate, handler = state._get_event_handler(event)
+        if not handler.is_background:
+            return None
+
+        async def _coro():
+            if self.event_namespace is None:
+                raise RuntimeError("App has not been initialized yet.")
+
+            # Process the event.
+            async for update in state._process_event(
```

Comment:
Should we abstract this to its own function?  Similar logic is repeated in the `app._process` as well as the upload logic

Reply:
this one is sort of hard to abstract away because each call site is slightly different:

* App._process_background.<locals>._coro: uses `state._process_event` and `emit_update` directly
* app.process: uses `state._process` and yields the updates
* upload: uses `state._process` and `emit_update` directly

Might be nice to try to unify these down the road, but I think I'll defer on this PR.